src/main/java/com/cdw/gatekeeper/exceptions/GlobalExceptionHandler.java

```diff
@@ -0,0 +1,88 @@
+package com.cdw.gatekeeper.exceptions;
+
+import com.cdw.gatekeeper.dto.FailureResponseDTO;
+import com.cdw.gatekeeper.utils.ExceptionUtility;
+import org.apache.coyote.BadRequestException;
+import org.springframework.http.HttpStatus;
+import org.springframework.http.ResponseEntity;
+import org.springframework.web.bind.annotation.ControllerAdvice;
+import org.springframework.web.bind.annotation.ExceptionHandler;
+import org.springframework.web.context.request.WebRequest;
+
+import java.io.IOException;
+
+/**
+ * GlobalExceptionHandler handles all exception thrown from the Gatekeeper controllers
+ */
+@ControllerAdvice
+public class GlobalExceptionHandler {
+
+    /**
+     * Handles resource not found exceptions
+     * @param ex
+     * @return ResponseEntity
+     */
+    @ExceptionHandler(ResourceNotFoundException.class)
+    public ResponseEntity<Object> handleResourceNotFoundException(ResourceNotFoundException ex) {
```

Comment:
response entity can be ignored, if RestControllerAdvice can be used and @ResponseStatus can be used to specify the status codes